Wikipedia/Code/wikipedia-language-variants.json

```diff
@@ -127,6 +127,15 @@
             "languageCode": "ku"
         }
     ],
+    "shi" : [
```

Comment:
@mazevedofs This file is not auto-generated, so you are correct to edit it directly.

This is the first time we've changed a language to support variants since 6.8.0, and I found some documentation that indicates it'll be a larger thing than adjusting this file. So I suggest backing out this change from this PR and making it it's own separate effort. Below are some of my notes for first places to start if you decide to tackle it:

### Add the Latin option to this file

Looks like this list will take over for all `shi` language options, so you will lose the original Latin option if you only have one here. Adding it in this array should fix it:

```
{
       "languageVariantCode": "shi-latn",
       "languageName": "Taclḥit",
       "localName": "Tachelhit (Latin)",
       "languageCode": "shi"
}
```

Note: I used [this json](https://www.mediawiki.org/w/api.php?action=query&format=json&meta=siteinfo&formatversion=2&siprop=languages%7Clanguagevariants&siinlanguagecode=en) to get the code value.

### Update `MediaWikiAcceptLanguageMapping.json`
 
Here's some nice [documentation](https://github.com/wikimedia/wikipedia-ios/blob/main/docs/languages.md) about this.

### Add a data migration step converting old persisted non-variant data to variants. Present onboarding modal informing them.

Here's some more documentation I found in this [file](https://github.com/wikimedia/wikipedia-ios/blob/main/WMF%20Framework/MWKDataStore%2BLanguageVariantMigration.swift) about this (see comment at the top).

Reply:
FYI - this file looked good but I think more will be needed from this comment to get these new variants working, so I'm backing this one out so that we can at least get [T331550](https://phabricator.wikimedia.org/T331550) in. Thanks!

Reply:
Oh I assumed that the core data migration would happen `automagically` with the corrected json, thanks for pulling it so we can get the other languages in!

Reply:
@mazevedofs Ah I'm sorry, you could be right, I assumed it wouldn't happen automatically. Happy to test this out in another PR. Note you'll probably also need to add the correct default variant choices for `shi` in `MediaWikiAcceptLanguageMapping.json` as well.